### PR TITLE
Convert resourceskus service to SDKv2

### LINKED ISSUE #3920: Update resourceskus service to SDK v2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts the resource skus service to azure-sdk-for-go version 2.

**Which issue(s) this PR fixes**:

Fixes #3920

**Special notes for your reviewer**:

This service doesn't provide "Create" or "Delete" functionality, it just lists and gets existing SKUs.

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
/retitle [WIP] Convert resourceskus service to SDKv2

There's a panic that shows up in a unit test I still haven't figured out.

**Comment by CecileRobertMichon** [comment]
is this ready for review @mboersma ? 

nit: PR description says role assignments, should be resource skus

**Comment by mboersma** [comment]
/retitle Convert resourceskus service to SDKv2

It's ready for review now I think. I had to work around another `panic()` similar to #3939.

File: azure/services/resourceskus/client.go [hunk 1]
```diff
@@ -19,55 +19,53 @@ package resourceskus
...
 // Client wraps go-sdk.
 type Client interface {
-	List(context.Context, string) ([]compute.ResourceSku, error)
+	List(context.Context, string) ([]armcompute.ResourceSKU, error)
 }
 
 // AzureClient contains the Azure go-sdk Client.
 type AzureClient struct {
-	skus compute.ResourceSkusClient
+	skus armcompute.ResourceSKUsClient
```
Comment (nojnhuh) [inline azure/services/resourceskus/client.go:35]: Should we make this a pointer for consistency?
```suggestion
	skus *armcompute.ResourceSKUsClient
```

Comment (mboersma) [reply azure/services/resourceskus/client.go:35]: Yes indeed, I must have missed this one.

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**